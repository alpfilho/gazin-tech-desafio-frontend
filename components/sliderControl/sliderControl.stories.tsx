import { ComponentMeta, ComponentStory } from '@storybook/react';

import { SliderControl } from './sliderControl';

export default {
	title: 'Components/Slider Control',
	component: SliderControl
} as ComponentMeta<typeof SliderControl>;

export const Left: ComponentStory<typeof SliderControl> = (args) => <SliderControl {...args} />;

Left.args = {
	show: true,
	type: 'left'
};

export const Right: ComponentStory<typeof SliderControl> = (args) => <SliderControl {...args} />;

Right.args = {
	show: true,
	type: 'right'
};
