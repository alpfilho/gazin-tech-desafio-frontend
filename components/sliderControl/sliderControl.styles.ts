import styled from 'styled-components';
import { motion } from 'framer-motion';

export const ControlContainer = styled(motion.button)<{ show: 'true' | undefined }>`
	z-index: 3;
	position: absolute;
	top: 0;
	bottom: 0;
	height: 100%;
	display: flex;
	justify-content: center;
	align-items: center;
	opacity: 0;
	cursor: ${({ show }) => (show ? 'pointer' : 'default')};
	pointer-events: ${({ show }) => (show ? 'all' : 'none')};
	border: none;
	padding: 0;

	will-change: transform, opacity;

	&:focus {
		outline: none;
		border: none;
	}
`;

export const LeftControlButton = styled(ControlContainer)`
	left: 0;
	right: auto;
	padding-left: 3em;
`;

export const RightControlButton = styled(ControlContainer)`
	right: 0;
	left: auto;
	padding-right: 3em;
`;

export const ControlIconContainer = styled.div`
	display: flex;
	justify-content: center;
	align-items: center;

	width: 5em;
	height: 5em;

	background-color: rgba(0, 0, 0, 0.66);
	border-radius: 50%;

	svg {
		height: 2em;
		width: auto;
	}
`;
