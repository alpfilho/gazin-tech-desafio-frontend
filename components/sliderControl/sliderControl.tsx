import React from 'react';
import { useTheme } from 'styled-components';
import { faChevronLeft, faChevronRight } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import {
	LeftControlButton,
	ControlIconContainer,
	RightControlButton
} from './sliderControl.styles';

interface SliderControlI {
	type: 'left' | 'right';
	controlFn: () => void;
}

export const SliderControl: React.FC<SliderControlI> = ({ type, controlFn }) => {
	const { colors } = useTheme();

	if (type === 'left') {
		return (
			<LeftControlButton onClick={controlFn}>
				<ControlIconContainer>
					<FontAwesomeIcon icon={faChevronLeft} color={colors.yellow} />
				</ControlIconContainer>
			</LeftControlButton>
		);
	} else {
		return (
			<RightControlButton onClick={controlFn}>
				<ControlIconContainer>
					<FontAwesomeIcon icon={faChevronRight} color={colors.yellow} />
				</ControlIconContainer>
			</RightControlButton>
		);
	}
};
