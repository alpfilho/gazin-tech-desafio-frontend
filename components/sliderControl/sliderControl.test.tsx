import { render, screen } from '@testing-library/react';
import { composeStories } from '@storybook/testing-react';

import * as stories from './sliderControl.stories';

const { Left, Right } = composeStories(stories);

describe('SliderControl', () => {
	it('render a left slider control', () => {
		render(<Left />);

		const sliderControl = screen.getByTestId('slider-control');
		expect(sliderControl).toBeInTheDocument();
	});

	it('left slider control render unchanged', () => {
		const { container } = render(<Left />);

		expect(container).toMatchSnapshot();
	});

	it('render a right slider control', () => {
		render(<Right />);

		const sliderControl = screen.getByTestId('slider-control');
		expect(sliderControl).toBeInTheDocument();
	});

	it('Right slider control render unchanged', () => {
		const { container } = render(<Right />);

		expect(container).toMatchSnapshot();
	});
});
