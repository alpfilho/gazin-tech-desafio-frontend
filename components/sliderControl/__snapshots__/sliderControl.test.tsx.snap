// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SliderControl Right slider control render unchanged 1`] = `
<div>
  <button
    class="ControlContainer-sc-ms3p6h-0 RightControlButton-sc-ms3p6h-2 CYGWr fehyoi"
    data-testid="slider-control"
  >
    <div
      class="ControlIconContainer-sc-ms3p6h-3 eAXRGx"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-chevron-right "
        color="#f3f1f1"
        data-icon="chevron-right"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M96 480c-8.188 0-16.38-3.125-22.62-9.375c-12.5-12.5-12.5-32.75 0-45.25L242.8 256L73.38 86.63c-12.5-12.5-12.5-32.75 0-45.25s32.75-12.5 45.25 0l192 192c12.5 12.5 12.5 32.75 0 45.25l-192 192C112.4 476.9 104.2 480 96 480z"
          fill="currentColor"
        />
      </svg>
    </div>
  </button>
</div>
`;

exports[`SliderControl left slider control render unchanged 1`] = `
<div>
  <button
    class="ControlContainer-sc-ms3p6h-0 LeftControlButton-sc-ms3p6h-1 CYGWr kbLFOO"
    data-testid="slider-control"
  >
    <div
      class="ControlIconContainer-sc-ms3p6h-3 eAXRGx"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-chevron-left "
        color="#f3f1f1"
        data-icon="chevron-left"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M224 480c-8.188 0-16.38-3.125-22.62-9.375l-192-192c-12.5-12.5-12.5-32.75 0-45.25l192-192c12.5-12.5 32.75-12.5 45.25 0s12.5 32.75 0 45.25L77.25 256l169.4 169.4c12.5 12.5 12.5 32.75 0 45.25C240.4 476.9 232.2 480 224 480z"
          fill="currentColor"
        />
      </svg>
    </div>
  </button>
</div>
`;
